File: easy/buddy-strings.py
class Solution:
    def buddyStrings(self, A, B):
        """
        :type A: str
        :type B: str
        :rtype: bool
        """
        if len(A) != len(B):
            return False
        
        if len(A) < 2:
            return False
        
        numA = collections.Counter(A)
        
        if numA != collections.Counter(B):
            return False
        
        diff = 0
        hasDup = False
        
        for a, b in zip(A, B):
            if numA[a] > 1:
                hasDup = True
            if a != b:
                diff += 1

        return diff == 2 or (hasDup and A == B)

File: easy/buddy-strings.swift
class Solution {
    func buddyStrings(_ A: String, _ B: String) -> Bool {
        if A.count != B.count {
            return false
        }
        
        var dictA = [Character: Int]()
        for c in A {
            dictA[c, default: 0] += 1
        }
        
        var dictB = [Character: Int]()
        for c in B {
            dictB[c, default: 0] += 1
        }
                
        if A == B {
            return dictA.count != A.count
        }
        
        if dictA == dictB {
            var arrA = Array(A)
            var diff = 0
            for (i, c) in B.enumerated() {
                if c != arrA[i] {
                    diff += 1
                }
            }
            if diff == 2 {
                return true
            }
        }
        
        return false
    }
}
